libdnf/rpm/solv_query.cpp

```diff
@@ -847,6 +929,58 @@ SolvQuery & SolvQuery::ifilter_epoch(libdnf::sack::QueryCmp cmp_type, const std:
     return *this;
 }
 
+SolvQuery & SolvQuery::ifilter_epoch(libdnf::sack::QueryCmp cmp_type, const std::vector<std::string> & patterns) {
+    bool cmp_not = (cmp_type & libdnf::sack::QueryCmp::NOT) == libdnf::sack::QueryCmp::NOT;
```

Comment:
Wouldn't this be sufficient?
`bool cmp_not = cmp_type & libdnf::sack::QueryCmp::NOT;`
Although it may fail on type conversion...

Reply:
Not really, `cmp_type & libdnf::sack::QueryCmp::NOT` returns `libdnf::sack::QueryCmp` type of object that is not convertible into bool (`enum class` vs. `enum` where conversion is available).